tests/integrationtests/DBusAsyncMethodsTests.cpp

```diff
@@ -201,6 +201,23 @@ TEST_F(SdbusTestObject, AnswersThatAsyncCallIsNotPendingAfterItHasBeenCompleted)
     ASSERT_TRUE(waitUntil([&call](){ return !call.isPending(); }));
 }
 
+TEST_F(SdbusTestObject, AsyncCallSupportsDefaultConstructionAndAssignment)
+{
+    // Default construct
+    sdbus::PendingAsyncCall call;
+
+    ASSERT_FALSE(call.isPending());
+
+    // Assign
+    call = m_proxy->doOperationClientSideAsync(100);
+
+    ASSERT_TRUE(call.isPending());
+
+    call.cancel();
+
+    ASSERT_FALSE(call.isPending());
+}
+
```

Comment:
I would suggest to split this into two test cases, each having a single responsibility, with names revealing behaviors under tests, and with more of "arrange-act-assert" ("given-when-then") structure:

```suggestion
TEST_F(SdbusTestObject, AnswersThatDefaultConstructedAsyncCallIsNotPending)
{
    sdbus::PendingAsyncCall call;
    
    ASSERT_FALSE(call.isPending());
}

TEST_F(SdbusTestObject, SupportsAsyncCallCopyAssignment)
{
    sdbus::PendingAsyncCall call;

    call = m_proxy->doOperationClientSideAsync(100);

    ASSERT_TRUE(call.isPending());
}

```

Reply:
Thanks, will do.